prov/efa/src/efa_mr.c

```diff
@@ -220,6 +255,57 @@ static struct fi_ops efa_mr_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static int efa_mr_shm_reg(struct efa_domain *efa_domain,
+			struct fi_mr_attr *attr, uint64_t flags,
+			struct efa_mr *mr)
+{
+	int ret;
+
+	ret = fi_mr_regattr(efa_domain->shm_domain, attr, flags,
+				&mr->shm_mr);
+	if (ret) {
+		FI_WARN(&rxr_prov, FI_LOG_MR,
+			"Unable to register shm MR buf (%s): %p len: %zu\n",
+			fi_strerror(-ret), attr->mr_iov->iov_base,
+			attr->mr_iov->iov_len);
+		fi_close(&mr->mr_fid.fid);
+		ofi_mr_map_remove(&efa_domain->util_domain.mr_map,
+					mr->mr_fid.key);
+		return ret;
+	}
+	return 0;
+}
+
+static int efa_mr_map_insert(struct efa_domain *efa_domain,
+			struct fi_mr_attr *attr, struct fi_mr_attr *shm_attr,
+			struct efa_mr *mr, struct fid_mr **mr_fid,
+			uint64_t flags)
```

Comment:
The function name is "efa_mr_map_insert", so it should not register shm MR.

Reply:
Makes sense, I'll have separate calls for the two functions.